infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -240,6 +243,47 @@ public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
         }
     }
 
+    public synchronized void refreshForSinkSideInitialization(LogReplicationSession sessionFromSource) {
```

Comment:
When the sink receives the registerSession rpc call, the correct way to handle it will be the logic in createIncomingSessionsBySubscriber. You don't need to construct the session, but will only need to add the sessions to the data structures and metadata tables (as in createIncomingSessionsBySubscriber).

the other sessions need not get affected, and also updating the topologyID is not needed.